src/multipart/multipart.rs

```diff
@@ -0,0 +1,469 @@
+extern crate uuid;
+
+use std;
+use std::borrow::Cow;
+use std::io::Read;
+use std::ascii::AsciiExt;
+use hyper::mime::Mime;
+
+/// A multipart/form-data request.
+#[derive(Debug)]
+pub struct MultipartRequest {
+    boundary: String,
+    fields: Vec<MultipartField>,
+}
+
+impl MultipartRequest {
+    /// Creates a new MultipartRequest without any content.
+    pub fn new() -> MultipartRequest {
+        MultipartRequest {
+            boundary: format!("{}", uuid::Uuid::new_v4().simple()),
+            fields: Vec::new(),
+        }
+    }
+    /// Adds a field, builder style.
+    pub fn field(mut self, field: MultipartField) -> MultipartRequest {
+        self.fields.push(field);
+        self
+    }
+    /// Adds multiple fields.
+    pub fn fields(&mut self, mut fields: Vec<MultipartField>) {
+        self.fields.append(&mut fields);
+    }
+    /// Turns this MultipartRequest into a RequestReader which implements the Read trait.
+    pub fn reader(self) -> RequestReader {
```

Comment:
This could be made private, since it's only used internally.

The way reqwest has done this in other modules is to have free public functions in the module (instead of methods on a public type). This allows usage in other reqwest modules, like `multipart::reader(req)`.